source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -134,25 +142,34 @@ void CheckRequestUtils::setHttpRequest(
 
   // Fill in the headers.
   auto* mutable_headers = httpreq.mutable_headers();
+  // Calling mutable_header_map() creates the field in the request; only do so when necessary.
+  auto* mutable_header_map = encode_raw_headers ? httpreq.mutable_header_map() : nullptr;
 
-  headers.iterate([request_header_matchers, mutable_headers](const Envoy::Http::HeaderEntry& e) {
+  headers.iterate([encode_raw_headers, request_header_matchers, mutable_headers,
```

Comment:
nit (I think): pass the request_header_matchers by ref to avoid shared-ptr reference counter increase.

Reply:
I don't think so. It's a reference to the shared pointer, not the shared pointer itself.